### PR TITLE
Add ShellCheck job

### PR DESCRIPTION
**Pull request by Tahler**
- Adds tools/run_shellcheck.sh which finds all shell scripts in the repository (including itself) and runs shellcheck against them.
- Adds a CircleCI job which runs this script

All current violations are ignored right now, so the new CircleCI job should always succeed unless someone violates a rule that's not currently excluded.

In subsequent PRs I will incrementally tackle the exclusions by enabling the check and fixing the violations.

/cc @mandarjog @costinm

### CODE REVIEW

File: tools/run_shellcheck.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+TOOLS_DIR="$(cd "$(dirname "${0}")" && pwd -P)"
+ISTIO_ROOT="$(cd "$(dirname "${TOOLS_DIR}")" && pwd -P)"
+
+EXCLUDES="1009,"
+EXCLUDES="${EXCLUDES}1020,"
```
Comment (mandarjog) [inline tools/run_shellcheck.sh]: can you comment on these numbers or a link to the documentation for these number?

Comment (Tahler) [reply tools/run_shellcheck.sh]: Done.

**Pull request merged**